src/InfluxDB2/WriteApi.php

```diff
@@ -103,7 +103,32 @@ public function writeRaw(string $data, string $precision = null, string $bucket
 
         $queryParams = ["org" => $orgParam, "bucket" => $bucketParam, "precision" => $precisionParam];
 
-        $this->post($data, "/api/v2/write", $queryParams);
+        $this->writeRawInternal($data, $queryParams, 0);
+    }
+
+    private function writeRawInternal(string $data, array $queryParams, int $attempts)
+    {
+        try {
+            $this->post($data, "/api/v2/write", $queryParams);
+        } catch (ApiException $e) {
+            $code = $e->getCode();
+
+            if ($code == null || ($code < 429 || $code > 503) || $attempts >= $this->writeOptions->maxRetries) {
```

Comment:
Only the `429` and `503` should be "retry-able".

```suggestion
            if ($code == null || !($code == 429 || $code == 503) || $attempts >= $this->writeOptions->maxRetries) {
```